lib/aws-record/record/batch_read.rb

```diff
@@ -0,0 +1,154 @@
+# Copyright 2015-2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not
+# use this file except in compliance with the License. A copy of the License is
+# located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions
+# and limitations under the License.
+
+module Aws
+  module Record
+    class BatchRead
+
+      BATCH_GET_ITEM_LIMIT = 100
+
+      def initialize(opts = {})
+        @client = opts[:client]
+      end
+
+      def find(klass, key = {})
+        unprocessed_key = format_unprocessed_key(klass, key)
+        store_unprocessed_key(klass, unprocessed_key)
+        store_item_class(klass, unprocessed_key)
+      end
+
+      def execute!
+        if unprocessed_keys.count > BATCH_GET_ITEM_LIMIT
+          operation_keys = unprocessed_keys[0..BATCH_GET_ITEM_LIMIT-1].dup
```

Comment:
I don't think the `dup` is necessary here.  The [..] should already return a new array.

Reply:
Done! I'll keep that in mind, thank you!